lisa/target.py

```diff
@@ -276,6 +276,8 @@ def __init__(self, kind, name='<noname>', tools=[], res_dir=None,
 
         logger.info('Effective platform information:\n{}'.format(self.plat_info))
 
+        self._init_extend_sched_events()
```

Comment:
We probably don't want to call that unconditionally, as just connecting to a target should work everywhere, including places where we cannot create/insert modules (lack of toolchain, lack of permissions, connecting to a target which is a container etc). 

My only thought for now is adding an `__init__` parameter (plus a `modules` config key under `kernel` sublevel) that would contain a list of kmodules to build/load. This way the user can choose which modules are needed, including custom ones, so we can add more of these if we want. 

@valschneider any thoughts on that ?

Reply:
I tried to do the module insertion in FtraceCollect.__init__() but I had issues with that, let me try again more seriously this time

Reply:
I had the same thought over the weekend: we can let FtraceCollector decide if that modules is needed depending on the events that are asked for. That would allow connecting to the target without uploading the module when there won't be any consumer for it.